### ISSUE TITLE
Change all instances of `call_later(.looping_call)` to `call_later(.looping_call().start`

### ISSUE THREAD

**Issue by ShadowJonathan** [matrix-org/synapse#10460]
```py
self.clock.call_later(
    30, self.clock.looping_call, run_timeout_handler, 5000
)
```

Changing this to the following would avoid a bit of hassle when debugging (see below);

```py
self.clock.call_later(
    30, self.clock.looping_call(run_timeout_handler, 5000).start
)
```

Because `DelayedCall` implicitly expects a normal function/method to be passed to `call_later`, the `__repr__` function errors on the following line;

https://github.com/twisted/twisted/blob/65f382f43dfe9bec9d25d99ce0cb05db115f5fac/src/twisted/internet/base.py#L241

`L.extend((" ", self.func.__qualname__, "("))`

Functions have a `__qualname__`, but object instances doesnt.

When `DelayedCall.debug` is true (when debugging the reactor), it calculates a `repr` of itself on init, this means that *all* `DelayedCall`s go past that above line, which means that any `DelayedCall`s waiting on `LoopingCall` will error out.

**Comment by erikjohnston** [comment]
```python
self.clock.call_later(
    30, self.clock.looping_call(run_timeout_handler, 5000).start
)
```

Doesn't make sense, as `self.clock.looping_call(run_timeout_handler, 5000)` will get called immediately and start the looping call. We'll then call `.start()` again 30s later which is a noop.

We seem to do this sort of thing in only two places and that's next to each other in the presence handler. I don't see *why* we're doing it there, probably someone got confused and thought that the looping call would be called immediately. So I'd just remove the `call_later` from those two lines entirely.


Seperately, I'm confused what the problem is. `self.clock.looping_call` should have a `__qualname__`? The API usage seems sane, so is this not a bug in Twisted when using `DelayedCall.debug`?

**Comment by ShadowJonathan** [comment]
> `self.clock.looping_call` should have a `__qualname__`

No that's exactly it, it doesn't. Class instances don't have `__qualname__`, class references and functions do, that's why twisted threw a hissy fit when I turned on `debug` for `DelayedCall`, which then tries to calculate `repr` up-front instead of lazily when called.

The twisted devs picked up on this and reverted a change that introduced this bug, and made it rc3 of the latest release. They also suggested to do this pattern instead, that's why I made the issue. `start` probably isn't  a noop, I'm gonna guess, it could have a side effect of starting that looping call, else the devs wouldn't have recommended that.

**Comment by erikjohnston** [comment]
> > `self.clock.looping_call` should have a `__qualname__`
> 
> No that's exactly it, it doesn't. Class instances don't have `__qualname__`, class references and functions do, that's why twisted threw a hissy fit when I turned on `debug` for `DelayedCall`, which then tries to calculate `repr` up-front instead of lazily when called.

Ooooh, its the *return* value that doesn't have a `__qualname__`.  Right, sorry.


> They also suggested to do this pattern instead, that's why I made the issue. `start` probably isn't a noop, I'm gonna guess, it could have a side effect of starting that looping call, else the devs wouldn't have recommended that.

`Clock` is a Synapse class, not a Twisted class, and so yes, `start` *is* a no-op and so your suggestion changes behaviour: https://github.com/matrix-org/synapse/blob/develop/synapse/util/__init__.py#L112

---

> The twisted devs picked up on this and reverted a change that introduced this bug, and made it rc3 of the latest release. 

So this is just a problem for 21.7.0rc1 and 21.7.0rc2 when using `DelayedCall`, and so will be fixed in 21.7? Given that I'm closing this as I don't think there is any point patching around a bug in an RC of an upstream library.
